File: 2.server/src/main/java/ru/vitaliy/petrov/server/forms/requests/documents/AdministrativeOffenceCaseInvestigationAddRequest.java
package ru.vitaliy.petrov.server.forms.requests.documents;

import lombok.Data;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import java.sql.Date;
import java.sql.Time;

@Data
public class AdministrativeOffenceCaseInvestigationAddRequest {

    @NotNull
    private Date dateOfFill;

    @NotNull
    private Time timeOfFill;

    @NotNull
    @NotEmpty
    private String placeOfFill;

    @NotNull
    private Long policeOfficerID;

    @NotNull
    private Long carAccidentEntityID;

    @NotNull
    @NotEmpty
    private String investigationReason;

    @NotNull
    @NotEmpty
    private String lawViolationInfo;

}

File: client/app/src/main/java/com/petrov/vitaliy/caraccidentapp/domain/models/requests/documents/AdministrativeOffenceCaseInvestigationAddRequest.kt
package com.petrov.vitaliy.caraccidentapp.domain.models.requests.documents

import java.sql.Date
import java.sql.Time

data class AdministrativeOffenceCaseInvestigationAddRequest(
    val dateOfFill: Date,
    val timeOfFill: Time,
    val placeOfFill: String,
    val policeOfficerID: Long,
    val carAccidentEntityID: Long,
    val investigationReason: String,
    val lawViolationInfo: String
)
